src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/DGSpecFactory.cs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Graph;
+using NuGet.Commands.Restore;
+using NuGet.Commands.Restore.Utility;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.ProjectModel;
+
+namespace NuGet.CommandLine.XPlat.Utility;
+
+internal class DGSpecFactory
+{
+    /// <summary>Create a <see cref="DependencyGraphSpec"/> for the target at the given path.</summary>
+    /// <param name="projectPath">A path to a solution or project file, or a directory where a solution or project exists.</param>
+    /// <returns>The full path to a solution or project file. Throws is none or more than one are found</returns>
+    /// <exception cref="ArgumentException">If no solution or project file is found, or if more than one is found.</exception>
+    /// <exception cref="ArgumentNullException">If projectPath is null or whitespace.</exception>
+    public static DependencyGraphSpec Create(string projectPath)
+    {
+        if (string.IsNullOrWhiteSpace(projectPath)) { throw new ArgumentNullException(nameof(projectPath)); }
+
+        IEnumerable<string> projects = MSBuildAPIUtility.GetListOfProjectsFromPathArgument(projectPath);
+
+        var globalProperties = new Dictionary<string, string>();
+
+        bool IsInnerBuild(ProjectGraphNode node)
+        {
+            return node.ProjectInstance.GlobalProperties.ContainsKey("TargetFramework");
+        }
+
+        var projectGraph = new ProjectGraph(projects, globalProperties, ProjectCollection.GlobalProjectCollection);
```

Comment:
I'm guessing this logic is similar to what static graph does correct?

We have a bit of a dependency challenge ofc, but I think it'll be a good idea to move this to NuGet.Commands eventually and share it among the 2 implementations.